### PR TITLE
fix: installation fails when global npm modules path does not exist

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: packages/backend/src/utils/npm.ts [hunk 1]
```diff
@@ -89,13 +97,9 @@ class Command {
   }
 
   private async getAccessResult(): Promise<string> {
-    const globalNodeModulesPath = await this.getGlobalNodeModulesPath();
-    // we assume that if custom path set by an user is writable
+    // we assume that if custom path set by an user it is writable
     if (_.isEmpty(customLocation.getPath())) {
```
Comment (alex-gilin) [inline packages/backend/src/utils/npm.ts:99]:
```suggestion
    if (!customLocation.getPath()) {
```

File: packages/backend/src/utils/npm.ts [hunk 2]
```diff
@@ -40,11 +40,19 @@ class Command {
...
+    void this.execCommand(`${NPM} root -g`).then((globalNodeModulesPath: string) => {
+      this.globalNodeModulesPathPromise = fs.promises
+        .mkdir(globalNodeModulesPath, { recursive: true })
+        .then(() => globalNodeModulesPath);
+    });
+  }
+
   private getGenLocationParams(): string {
     const customInstallationPath = customLocation.getPath();
     return _.isEmpty(customInstallationPath) ? "-g" : `--prefix ${customInstallationPath}`;
```
Comment (alex-gilin) [inline packages/backend/src/utils/npm.ts:56]:
```suggestion
    return !customInstallationPath ? "-g" : `--prefix ${customInstallationPath}`;
```

File: packages/backend/src/utils/npm.ts [hunk 3]
```diff
@@ -40,11 +40,19 @@ class Command {
...
     this.SET_DEFAULT_LOCATION = messages.set_default_location(customLocation.DEFAULT_LOCATION);
     this.isInBAS = !_.isEmpty(_.get(process, "env.WS_BASE_URL"));
   }
 
+  private setGlobalNodeModulesPath() {
+    void this.execCommand(`${NPM} root -g`).then((globalNodeModulesPath: string) => {
+      this.globalNodeModulesPathPromise = fs.promises
+        .mkdir(globalNodeModulesPath, { recursive: true })
+        .then(() => globalNodeModulesPath);
+    });
```
Comment (alex-gilin) [inline packages/backend/src/utils/npm.ts]: what is an advantage of using that calling method vs await mkdir(...) ?

File: packages/backend/src/utils/npm.ts [hunk 4]
```diff
@@ -112,8 +116,8 @@ class Command {
   }
 
   private isPathWritable(path: string): Promise<boolean> {
-    return promises
-      .access(path, constants.W_OK)
+    return fs.promises
+      .access(path, fs.constants.W_OK)
```
Comment (alex-gilin) [inline packages/backend/src/utils/npm.ts:118]: same as above ..

File: packages/backend/src/yeomanui.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import * as path from "path";
-import { mkdirs, readdir, stat } from "fs-extra";
+import * as fs from "fs";
```
Comment (alex-gilin) [inline packages/backend/src/yeomanui.ts:2]:
```suggestion
import * as fs from "fs/promises";
```

File: packages/backend/src/yeomanui.ts [hunk 6]
```diff
@@ -185,9 +185,9 @@ export class YeomanUI {
     const result = { targetFolderPath: folderPath, childDirs };
 
     try {
-      for (const file of await readdir(folderPath)) {
+      for (const file of await fs.promises.readdir(folderPath)) {
```
Comment (alex-gilin) [inline packages/backend/src/yeomanui.ts:188]:
```suggestion
      for (const file of await fs.readdir(folderPath)) {
```

File: packages/backend/src/yeomanui.ts [hunk 7]
```diff
@@ -185,9 +185,9 @@ export class YeomanUI {
...
         const resourcePath: string = path.join(folderPath, file);
-        if ((await stat(resourcePath)).isDirectory()) {
+        if ((await fs.promises.stat(resourcePath)).isDirectory()) {
```
Comment (alex-gilin) [inline packages/backend/src/yeomanui.ts:190]:
```suggestion
        if ((await fs.stat(resourcePath)).isDirectory()) {
```

File: packages/backend/src/yeomanui.ts [hunk 8]
```diff
@@ -213,7 +213,7 @@ export class YeomanUI {
         this.setCwd(targetFolderProp);
       }
       const targetFolder = this.getCwd();
-      await mkdirs(targetFolder);
+      await fs.promises.mkdir(targetFolder, { recursive: true });
```
Comment (alex-gilin) [inline packages/backend/src/yeomanui.ts:216]:
```suggestion
      await fs.mkdir(targetFolder, { recursive: true });
```

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: packages/backend/src/utils/npm.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts:99]: No change here

File: packages/backend/src/utils/npm.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts:56]: No change here

File: packages/backend/src/utils/npm.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts]: explained

File: packages/backend/src/utils/npm.ts (hunk 4 above)
Comment (slavik-lvovsky) [reply packages/backend/src/utils/npm.ts:118]: No change here

File: packages/backend/src/yeomanui.ts (hunk 5 above)
Comment (slavik-lvovsky) [reply packages/backend/src/yeomanui.ts]: Done

File: packages/backend/src/yeomanui.ts (hunk 7 above)
Comment (slavik-lvovsky) [reply packages/backend/src/yeomanui.ts]: Done

File: packages/backend/src/yeomanui.ts (hunk 8 above)
Comment (slavik-lvovsky) [reply packages/backend/src/yeomanui.ts]: Done

**Comment by slavik-lvovsky** [comment]
ready for another review cycle

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**